lib/src/sv_sign.c

```diff
@@ -918,12 +924,11 @@ signed_video_set_private_key(signed_video_t *self, const char *private_key, size
     SV_THROW(openssl_private_key_malloc(self->sign_data, private_key, private_key_size));
     SV_THROW(openssl_read_pubkey_from_private_key(self->sign_data, &self->pem_public_key));
 
-    self->plugin_handle = sv_signing_plugin_session_setup(private_key, private_key_size);
-    SV_THROW_IF(!self->plugin_handle, SV_EXTERNAL_ERROR);
-    // If ONVIF is available, call initialize Onvif.
-    if (self->onvif) {
-      SV_THROW(initialize_onvif(self));
-    }
+    // Fallback to signing plugin if ONVIF is unavailable or ONVIF cant be initialized.
+    if(!self->onvif || !(initialize_onvif(self) == SV_OK)) {
```

Comment:
This is a really awkward check and construction. Have you tested this for both orders of "set_private_key" and "set_attestation"?
Have you checked how many signing plugins run when signing has started?